packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue

```diff
@@ -83,37 +83,27 @@
               :key="`public-link-edit-option-${i}`"
               class="oc-rounded oc-menu-item-hover"
             >
-              <oc-datepicker
-                v-if="option.showDatepicker"
-                v-model="newExpiration"
-                class="link-expiry-picker oc-flex oc-width-1-1"
-                :min-date="expirationRules.min"
-                :max-date="expirationRules.max"
-                :locale="$language.current"
-                :is-required="expirationRules.enforced"
-              >
-                <template #default="{ togglePopover }">
-                  <oc-button
-                    :data-testid="`files-link-id-${linkShare.id}-edit-${option.id}`"
-                    appearance="raw"
-                    class="oc-p-s action-menu-item"
-                    :variation="option.variation"
-                    @click="togglePopover"
-                  >
-                    <oc-icon :name="option.icon" fill-type="line" size="medium" />
-                    <span v-text="option.title" />
-                  </oc-button>
-                  <oc-button
-                    v-if="option.remove && option.remove.isRemovable"
-                    :data-testid="`files-link-id-${linkShare.id}-edit-${option.id}`"
-                    :aria-label="option.remove.title"
-                    appearance="raw"
-                    @click="option.remove.method"
-                  >
-                    <oc-icon :name="option.remove.icon" />
-                  </oc-button>
-                </template>
-              </oc-datepicker>
+              <div v-if="option.showDatepicker" class="oc-flex">
+                <oc-button
```

Comment:
I think at least for screen readers we need a hint that using this button lets the user edit the expiration date. Debatable if we also need it for non-screen reader users. Maybe it's just simpler to use `Edit expiration` as button label instead of the current expiration.

Reply:
Follow up